juspay/hyperswitch
# feat(router): add refunds manual-update api

## Type of Change



## Description
add refunds manual-update api

refunds manual-update api allows to update the refund details like status, error code and error message


### Additional Changes





## Motivation and Context



## How did you test it?

Tested Manually
1. Create a refund
![image](https://github.com/juspay/hyperswitch/assets/56996463/fd55289e-a729-4e4b-9c91-c8ade86b7eea)

2. Check the status, error code and message in DB
![image](https://github.com/juspay/hyperswitch/assets/56996463/d9103b5b-ad9e-48b9-842b-04ed2d70fc35)

3. Call Manual Update API for refund
CURL
```
curl --location --request PUT '{{API_BASE_URL}}/refunds/{{refund_id}}/manual-update' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: {{ADMIN_API_KEY}}' \
--data-raw '{
    "merchant_id": "{{MERCHANT_ID}}",
    "status": "failed",
    "error_code": "code",
    "error_message": "message"
}'
```
![image](https://github.com/juspay/hyperswitch/assets/56996463/7266777f-e1b3-4f75-851c-87d6b8036301)

4. Observe the changes in DB
![image](https://github.com/juspay/hyperswitch/assets/56996463/79dc7798-c3d5-4efe-8e83-8764e6d47777)

## Checklist
